src/Assignments/ProblemIframe.tsx

```diff
@@ -36,18 +36,18 @@ export const ProblemIframe: React.FC<ProblemIframeProps> = ({
     const [error, setError] = useState('');
     const [lastSubmission, setLastSubmission] = useState({});
     const height = '100vh';
-    let isClean = false;
 
     const { setLastSavedAt, setLastSubmittedAt } = useCurrentProblemState();
 
     useEffect(()=>{
-        setLoading(true);
         // We need to reset the error state since a new call means no error
         setError('');
         // If you don't reset the rendered html you won't get the load event
         // Thus if you go to an error state and back to the success state
         // The rendered html will never call load handler which will never stop loading
         setRenderedHTML('');
+        // srcdoc='' triggers onLoad with setLoading(false) so setLoading(true) isn't effective until now
```

Comment:
srcdoc should trigger onLoad regardless, I'm not sure of the benefit of moving this down but if it works it works, we can talk about it later

Reply:
right, when srcdoc = '' triggers onLoad, then onLoad does setLoading(false), and then the renderedHTML gets updated with the actual problem contents, meanwhile setLoading remains false - which is why we see that the GeoGebra problems aren't being hidden while they load.

Reply:
Great catch!!!